h/cli/commands/authclient.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+import click
+
+from h import models
+
+
+@click.group()
+def authclient():
+    """Manage OAuth clients."""
+
+
+@authclient.command()
+@click.option('--name', prompt=True, help="The name of the client")
+@click.option('--authority', prompt=True, help="The authority (domain name) for the client")
+@click.pass_context
+def add(ctx, name, authority):
+    """
+    Create a new OAuth client.
+
+    Create a new OAuth client for a "publisher" account, which can be used to
+    create and manage resources within a given namespace ("authority").
+    """
+    request = ctx.obj['bootstrap']()
+
+    authclient = models.AuthClient(name=name, authority=authority)
+    request.db.add(authclient)
+    request.db.flush()
+
+    id_ = authclient.id
+    secret = authclient.secret
+
+    request.tm.commit()
+
+    click.echo('OAuth client for {authority} created\n'
+               'Client ID: {id}\n'
+               'Client Secret: {secret}'.format(authority=authority, id=id_,
+                                                secret=secret))
+
+
+@authclient.command()
+@click.argument('authority')
+@click.pass_context
+def secret(ctx, authority):
+    """
+    Display client ID and secret.
+
+    Display the ID and secret for an OAuth client associated with a given
```

Comment:
Strictly speaking the model permits multiple OAuth clients for a given authority. For practical purposes however, we're only going to have one per authority as far as I understand.